src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -528,18 +568,60 @@ def getMSOutputTransferInfo(self, wflow):
             wflow['TransferDone'] = True
         return wflow
 
-    def getRucioRules(self, wflow, gran, rucioAcct):
+    def setParentDatasets(self, wflow):
+        """
+        Used to resolve parent datasets for a workflow.
+        :param  wflow:   A MSRuleCleaner workflow representation
+        :return:         The workflow object
+        """
+        if wflow['InputDataset'] and wflow['IncludeParents']:
+            childDataset = wflow['InputDataset']
+            parentDataset = findParent([childDataset], self.msConfig['dbsUrl'])
+            if parentDataset:
+                wflow['ParentDataset'].append(parentDataset[childDataset])
+                wflow['ParentDataset'] = list(set(wflow['ParentDataset']))
+                msg = "Found parent %s for input dataset %s for workflow: %s "
+                self.logger.debug(msg, parentDataset, wflow['InputDataset'], wflow['RequestName'])
```

Comment:
I suggest to convert this log level to INFO. It's very rare to have such workflows, and it's an important piece of information.

Reply:
Done